cylc/flow/workflow_files.py

```diff
@@ -1683,14 +1684,15 @@ def install_workflow(
     source_link = cylc_install.joinpath(WorkflowFiles.Install.SOURCE)
     # check source link matches the source symlink from workflow dir.
     cylc_install.mkdir(parents=True, exist_ok=True)
-    if not source_link.exists():
-        if source_link.is_symlink():
-            raise WorkflowFilesError(
-                f'Workflow source dir is not accessible:'
-                f' "{source_link.resolve()}".\n'
-                f'Restore the source or modify the "{source_link}" symlink'
-                ' to continue.'
-            )
+    if not source_link.exists() and source_link.is_symlink():
+        # Condition represents a broken symlink.
+        raise WorkflowFilesError(
+            f'Workflow source dir is not accessible:'
+            f' "{source_link.resolve()}".\n'
+            f'Restore the source or modify the "{source_link}" symlink'
+            ' to continue.'
+        )
+    elif not source_link.exists():
         install_log.info(f"Creating symlink from {source_link}")
         source_link.symlink_to(source.resolve())
     else:
```

Comment:
The error message below could do with some work too. Something like
```diff
diff --git a/cylc/flow/workflow_files.py b/cylc/flow/workflow_files.py
index aab1700d2..61290966b 100644
--- a/cylc/flow/workflow_files.py
+++ b/cylc/flow/workflow_files.py
@@ -1694,9 +1694,11 @@ def install_workflow(
         install_log.info(f"Creating symlink from {source_link}")
         source_link.symlink_to(source.resolve())
     else:
-        if not source_link.resolve() == source.resolve():
+        if source_link.resolve() != source.resolve():
             raise WorkflowFilesError(
-                "Source directory not consistent between runs.")
+                f"Cannot install from {source} as it does not match "
+                f"existing workflow source: {source_link.resolve()}"
+            )
         install_log.info(
             f'Symlink from "{source_link}" to "{source}" in place.')
     install_log.info(f'INSTALLED {named_run} from {source}')
```

Reply:
Will have a poke at this